src/PhpDoc/TypeNodeResolver.php

```diff
@@ -474,6 +475,34 @@ private function resolveGenericTypeNode(GenericTypeNode $typeNode, NameScope $na
 			}
 
 			return new ErrorType();
+		} elseif ($mainTypeName === 'int') {
+			if (count($genericTypes) === 2) { // int<min, max>, int<1, 3>
+
+				$minType = $genericTypes[0];
+				$maxType = $genericTypes[1];
+
+				if ($minType instanceof ConstantIntegerType) {
+					$min = $minType->getValue();
+				} elseif ($minType instanceof ObjectType) {
+					if ($minType->getClassName() === "PHPStan\Generics\GenericClassStringType\min") {
```

Comment:
This doesn't make sense, it's not going to work in a different namespace. Try looking for IdentifierTypeNode in `$typeNode->genericTypes` instead.

Reply:
lol, after reading your comment, I realized why the class-name contained this PHPStan prefixed string, which I initially thought of is a magic-value ;-)

changed the impl to use `IdentifierTypeNode` and added a few more testcases.

thanks again for your input